include/downloader.h

```diff
@@ -194,7 +194,7 @@ class Downloader
    * @param options: A series of pair <option_name, option_value> to pass to aria.
    * @return: The newly created Download.
    */
-  std::shared_ptr<Download> startDownload(const std::string& uri, const Options& options = {});
+  std::shared_ptr<Download> startDownload(const std::string& uri, const std::string& downloadDir, Options options = {});
```

Comment:
This change deserves a commit of its own - the description of the change `Remove getDataDirectory` doesn't cover it at all. In that commit, please also stop passing the `--dir` option to the `aria2c` command line (in the `Aria2` constructor).